### PR TITLE
Solve None error in json dump

### PR DESCRIPTION
**Pull request by harshitbansal05**
Presently, when a script is created or edited via `new_json` or `edit_json`, `None` is appended at the end. It is due to the line `output_file.write(json_str + '\n')`, where `json_str` is None. Currently, `json_str = str(json.dump(contents, output_file, sort_keys=True, indent=4, separators=(',', ': ')))` is called, but `json.dump()` itself writes the json formatted stream to the file. Thus, it returns None (which eventually gets assigned to `json_str`). This pull request aims to fix this. Please correct me if I am wrong.

### CODE REVIEW

**Comment by henrykironde** [comment]
We solved something like this recently, Are you using python3?

**Comment by harshitbansal05** [comment]
yes, @henrykironde

**Comment by henrykironde** [comment]
Okey will test it though it looks fine. We shall be done by tomorrow.

File: retriever/lib/datapackage.py [hunk 1]
```diff
@@ -247,9 +247,9 @@ def create_json():
     file_name = contents['name'] + ".json"
     file_name = file_name.replace('-', '_')
     with open(os.path.join(HOME_DIR, 'scripts', file_name), 'w') as output_file:
-        json_str = str(json.dump(contents, output_file, sort_keys=True, indent=4,
-                              separators=(',', ': ')))
-        output_file.write(json_str + '\n')
+        json.dump(contents, output_file, sort_keys=True, indent=4,
+                  separators=(',', ': '))
+        output_file.write('\n')
         print("\nScript written to " + file_name)
```
Comment (henrykironde) [inline retriever/lib/datapackage.py]: looks fine. 
Could you change `for key, val in obj.items():` to `for key, val in obj.copy().items():`
https://github.com/weecology/retriever/pull/1306/files#diff-7cb8d9949f19aa110e816b90e44ffb2dR261

Comment (harshitbansal05) [reply retriever/lib/datapackage.py]: Done.

**Comment by henrykironde** [comment]
Thanks @harshitbansal05 👍

**Pull request merged**